src/WebJobs.Extensions.DurableTask/DurableOrchestrationContext.cs

```diff
@@ -398,6 +366,97 @@ public void ContinueAsNew(object input)
             this.ContinuedAsNew = true;
         }
 
+        internal async Task<TResult> CallDurableTaskFunctionAsync<TResult>(string functionName,
+            Extensions.DurableTask.RetryOptions retryOptions,
+            object[] parameters)
+        {
+            this.ThrowIfInvalidAccess();
+
+            // TODO: Support for versioning
+            string version = DefaultVersion;
+            FunctionType functionType = this.config.GetFunctionType(functionName, version);
+
+            Task<TResult> callTask;
+
+            switch (functionType)
+            {
+                case FunctionType.Activity:
+                    if (retryOptions == null)
+                    {
+                        callTask = this.innerContext.ScheduleTask<TResult>(functionName, version, parameters);
+                    }
+                    else
+                    {
+                        callTask = this.innerContext.ScheduleWithRetry<TResult>(functionName, version,
+                            retryOptions.GetRetryOptions(), parameters);
+                    }
+                    break;
+                case FunctionType.Orchestrator:
+                    callTask = this.innerContext.CreateSubOrchestrationInstance<TResult>(functionName, version,
```

Comment:
What if someone calls `CallFunctionWithRetryAsync` for an orchestrator function and includes a `RetryOptions` parameter? I see that it gets ignored in this implementation, but do you think that behavior is going to be acceptable from a user's perspective? Had you considered throwing an exception? Or do you think we need to revisit reusing CallFunctionAsync for both activity and orchestrator functions since they now have different applicable parameter sets?

Reply:
I've added support for orchestrator functions as well.

---
In reply to: [137793230](https://github.com/Azure/azure-functions-durable-extension/pull/51#discussion_r137793230) [](ancestors = 137793230)

Reply:
Oh, I didn't realize this was already supported! Perfect!

---
In reply to: [137895273](https://github.com/Azure/azure-functions-durable-extension/pull/51#discussion_r137895273) [](ancestors = 137895273,137793230)